src/main/java/card/CardGenerator.java

```diff
@@ -0,0 +1,64 @@
+package card;
+
+import java.util.Random;
+
+public class CardGenerator {
+
+    private final Random random;
+
+    public CardGenerator() {
+        random = new Random();
+    }
+
+    public Card generateCredentials() {
+        return new Card(generateCard(), generatePIN());
+    }
+
+    public boolean isCorrectCardNumber(String cardNumber) {
```

Comment:
The validation should be outside of card generation.

Reply:
Exported validation to the `BankingSystem` class